examples/bsc-p2p/src/block_import/service.rs

```diff
@@ -0,0 +1,143 @@
+    pub fn new() -> (Self, UnboundedReceiver<IncomingBlock<T>>, UnboundedSender<Outcome<T>>) {
+        let (to_import, from_network) = mpsc::unbounded_channel();
+        let (to_network, import_outcome) = mpsc::unbounded_channel();
+
+        (Self { to_import, import_outcome }, from_network, to_network)
+    }
+
+    /// Sends the block to import to the service
+    pub fn send_block(
+        &self,
+        block: BlockMsg<T>,
+        peer_id: PeerId,
+    ) -> Result<(), Box<dyn std::error::Error>> {
+        self.to_import.send((block, peer_id))?;
+        Ok(())
+    }
+
+    /// Poll for the next import outcome
+    pub fn poll_outcome(&mut self, cx: &mut Context<'_>) -> Poll<Option<Outcome<T>>> {
+        self.import_outcome.poll_recv(cx)
+    }
+}
+
+type ImportFuture = Pin<Box<dyn Future<Output = ()> + Send>>;
+
+pub(crate) struct ImportService<T: EngineTypes> {
+    /// The handle to communicate with the engine service
+    engine: BeaconConsensusEngineHandle<T>,
+    /// Receive the new block from the network
+    from_network: UnboundedReceiver<IncomingBlock<T>>,
+    /// Send the outcome of the import to the network
+    to_network: UnboundedSender<Outcome<T>>,
+    /// Pending block imports.
+    pending_imports: FuturesUnordered<ImportFuture>,
+}
+
+impl<T: EngineTypes> ImportService<T> {
+    /// Create a new block import service
+    pub(crate) fn new(engine: BeaconConsensusEngineHandle<T>) -> (Self, ImportHandle<T>) {
+        let (handle, from_network, to_network) = ImportHandle::new();
+        (
+            Self { engine, from_network, to_network, pending_imports: FuturesUnordered::new() },
+            handle,
+        )
+    }
+
+    /// Add a new block import task to the pending imports
+    fn import(&mut self, block: BlockMsg<T>, peer_id: PeerId) {
+        let engine = self.engine.clone();
+        let to_network = self.to_network.clone();
+
+        let fut = async move {
+            let sealed_block = block.block.block.clone().seal();
+            let payload = T::block_to_payload(sealed_block);
+            let payload_result = engine.new_payload(payload).await;
+            let outcome = match payload_result {
+                Ok(payload_status) => match payload_status.status {
+                    PayloadStatusEnum::Valid => Outcome::<T> {
+                        peer: peer_id,
+                        result: Ok(BlockValidation::ValidBlock { block }),
+                    },
+                    PayloadStatusEnum::Invalid { validation_error } => Outcome::<T> {
+                        peer: peer_id,
+                        result: Err(BlockImportError::InvalidPayload(validation_error)),
+                    },
+                    _ => return,
+                },
+                Err(err) => {
+                    Outcome::<T> { peer: peer_id, result: Err(BlockImportError::Engine(err)) }
+                }
+            };
+
+            // TODO: FCU handling and send to engine
```

Comment:
this we need to track separately, like when do we decide to issue a fcu, e.g. by monitoring the height but we can do this separately